druid/src/main/java/org/apache/calcite/adapter/druid/DruidConnectionImpl.java

```diff
@@ -350,24 +350,24 @@ private void parseField(List<String> fieldNames, List<ColumnMetaData.Rep> fieldT
           if (s.equals("Infinity") || s.equals("-Infinity") || s.equals("NaN")) {
             throw new RuntimeException("/ by zero");
           }
+          rowBuilder.set(i, s);
+          break;
         case FLOAT:
         case PRIMITIVE_FLOAT:
         case PRIMITIVE_DOUBLE:
         case NUMBER:
         case DOUBLE:
           if (s.equals("Infinity")) {
             rowBuilder.set(i, Double.POSITIVE_INFINITY);
-            break;
           } else if (s.equals("-Infinity")) {
             rowBuilder.set(i, Double.NEGATIVE_INFINITY);
-            break;
           } else if (s.equals("NaN")) {
             rowBuilder.set(i, Double.NaN);
-            break;
+          } else {
+            rowBuilder.set(i, s);
           }
-          //fallthrough
+          break;
         default:
-          rowBuilder.set(i, s);
```

Comment:
I modified your code so that exceptional conditions throw or return, and non-exceptional conditions all go through the same 'rowBuilder.set(i, s)'. See https://github.com/julianhyde/calcite/commit/cbbe627feac1293796f68ff8ec8d4bd7b64724d2, and let me know if there's a problem. Otherwise I'll commit shortly.